extensions/lua/LuaScriptException.h

```diff
@@ -15,23 +15,26 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+#pragma once
 
-#include <memory>
+#include <exception>
 #include <utility>
 #include <string>
+#include <stdexcept>
 
-#include "PyOutputStream.h"
+namespace org::apache::nifi::minifi::extensions::lua {
 
-#include "utils/gsl.h"
+class LuaScriptException : public std::exception {
+ public:
+  explicit LuaScriptException(std::string error) : error_(std::move(error)) {}
 
-namespace org::apache::nifi::minifi::python {
+  virtual ~LuaScriptException() noexcept = default;
 
-PyOutputStream::PyOutputStream(std::shared_ptr<io::OutputStream> stream)
-    : stream_(std::move(stream)) {
-}
+  virtual const char * what() const noexcept {
+    return error_.c_str();
+  }
```

Comment:
as far as I know, an copying an `std::exception` must not throw, and `std::string` does not guarantee that, I think it would be safer to inherit from `std::runtime_error` which does some reference counting in the background

Reply:
Good catch, changed them in https://github.com/apache/nifi-minifi-cpp/pull/1504/commits/4a89cb33d7af056a2ef1a14dacd6cb5cd1cde02f